### PR TITLE
added hover version for (almost) all buttons & Fix #menuCloseButton & fixed applying dark theme

### PR DESCRIPTION
**Pull request by giwih**
## Description
Now all buttons and links have hover effect, “close” doesn't go anywhere

https://github.com/user-attachments/assets/b81c39bb-831f-44a4-98b4-08438617f35f

## Related Issues
- Issue #278 

## Checklist
- [x] I have read and followed the [Contributing Guidelines](../CONTRIBUTING.md).
- [x] I have added necessary tests.
- [x] The project builds and runs without issues.

### CODE REVIEW

**Comment by prem-k-r** [comment]
Looks great!
Can you also check for the dark theme? It may require defining colors separately.
Dark theme styling is in the script.js file

File: style.css [hunk 1]
```diff
@@ -1511,6 +1511,11 @@ body {
 	display: flex;
 	align-items: center;
 	text-decoration: none;
+	transition: background-color 0.3s ease;;
```
Comment (prem-k-r) [inline style.css:1514]: double semi colon

File: script.js [hunk 2]
```diff
@@ -1170,140 +1170,164 @@ const applySelectedTheme = (colorValue) => {
...
                 fill: #bfbfbf;
             }
 
-            .menuicon{
+            .dark-theme .menuicon{
                 color: #c2c2c2;
             }
 
-            #menuButton::before{
+            .dark-theme#menuButton::before{
```
Comment (prem-k-r) [inline script.js:1301]: space missing before #

**Comment by prem-k-r** [comment]
Great work!

There is a separate issue when using a theme other than the default blue one: upon page reload, the search text's bg-color, mic icon, and radio buttons for search engines briefly flash in blue for a split second.
If you have time and can find a fix, that would be awesome.

There is also an another significant issue occurs when opening a new tab (with or without a non-default blue theme): the blue theme, along with both the analog and digital clock, briefly flashes for a split second.

**Comment by prem-k-r** [comment]
@XengShi please review this at your convenience. 
This fixes the issue of dark theme colors persisting even after theme change

**Comment by giwih** [comment]
> There is also an another significant issue occurs when opening a new tab (with or without a non-default blue theme): the blue theme, along with both the analog and digital clock, briefly flashes for a split second.

On slow motion you can see that sometimes when you open a new tab (and when you reload too) you see a blue theme, I think this can be fixed but it may take a little longer, as a temporary solution you can slow down the page loading by a couple milliseconds or add a black screen for a couple milliseconds, but I don't think this is a good idea and for now it's better to leave it like this


https://github.com/user-attachments/assets/9f9424da-16ba-4f14-8e8d-796b62d82707

**Comment by giwih** [comment]
in https://github.com/XengShi/materialYouNewTab/pull/304/commits/6fa436aa7d7e677ced20d260a3115ce1aa1cabb8 fixed search bg-color, mic icon, and radio buttons flashing for search engines briefly*

**Review by prem-k-r — APPROVED** [review]

**Pull request merged**